features/step_definitions/api/application_plans_steps.rb

```diff
@@ -0,0 +1,291 @@
+  visit admin_service_application_plans_path(default_service)
+end
+
+Then "they can add new application plans" do
+  click_link 'Create Application plan'
+  fill_in('Name', with: 'Basic')
+  click_on('Create Application Plan', wait: 5)
+
+  assert_content /created application plan basic/i
+  assert current_path, admin_service_application_plans_path(default_service)
+end
+
+When "an admin selects the action copy of an application plan" do
+  @plan = FactoryBot.create(:application_plan, issuer: default_service)
+
+  visit admin_service_application_plans_path(default_service)
+  find_action_for_plan(/copy/i, @plan).click
+end
+
+When "an admin clicks on an application plan" do
+  @plan = FactoryBot.create(:application_plan, issuer: default_service, name: "Old name")
+
+  visit admin_service_application_plans_path(default_service)
+  click_link @plan.name
+end
+
+Then "an admin can delete it from the application plans page" do
+  visit admin_service_application_plans_path(default_service)
+  delete_plan_from_table_action(@plan)
+end
+
+When "an application plan {is} being used in an(y) application(s)" do |used|
+  @plan = FactoryBot.create(:application_plan, service: default_service)
+  FactoryBot.create(:cinstance, service: default_service, plan: @plan) if used
+
+  assert_equal used, @plan.reload.contracts_count.positive?
+end
+
+Then "an admin cannot delete it from the application plans page" do
+  visit admin_service_application_plans_path(default_service)
+  assert_not find_action_for_plan(/delete/i, @plan)
+end
+
+When "an admin hides a plan from the application plans page" do
+  @plan = FactoryBot.create(:published_plan, issuer: default_service)
+  visit admin_service_application_plans_path(default_service)
+
+  hide_plan_and_assert(@plan)
+end
+
+When "an admin publishes a plan from the application plans page" do
+  @plan = FactoryBot.create(:application_plan, issuer: default_service)
+  visit admin_service_application_plans_path(default_service)
+  publish_plan_and_assert(@plan)
+end
+
+Then "a buyer {will} be able to use it for their applications" do |will|
+  assert_equal will, @plan.service.application_plans.not_custom.published.include?(@plan)
+end
+
+When "an admin is looking for an application plan" do
+  ApplicationPlan.destroy_all
+  @plan_a = FactoryBot.create(:application_plan, issuer: default_service, name: 'This is number One')
+  @plan_b = FactoryBot.create(:application_plan, issuer: default_service, name: 'Now the second one')
+  @plan_c = FactoryBot.create(:application_plan, issuer: default_service, name: 'Finally the Last')
+
+  FactoryBot.create(:buyer_account, provider_account: @provider, org_name: 'Org 1').buy!(@plan_a)
+  FactoryBot.create(:buyer_account, provider_account: @provider, org_name: 'Org 2').buy!(@plan_b)
+
+  @plan_b.publish!
+  @plan_c.publish!
+
+  visit admin_service_application_plans_path(default_service)
+  assert_plans_table [@plan_a, @plan_b, @plan_c]
+end
+
+#TODO: FIXME: rename to: application plan
```

Comment:
We should finish this #TODO before merging I guess.

Reply:
It doesn't really belong to this PR, it would introduce unwanted changes IMO.